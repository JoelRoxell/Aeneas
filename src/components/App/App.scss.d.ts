// This file is generated automatically
export const app: string;
export const view: string;
export const version: string;
