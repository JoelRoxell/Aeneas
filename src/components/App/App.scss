* {
  box-sizing: border-box;
  font-family: 'Open sans', sans-serif;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

html,
body {
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  font-size: $rem-base + px;
}

:global(#root) {
  height: 100%;
}

.app {
  display: flex;
  height: 100%;
  flex-direction: column;
  overflow: hidden;

  .view {
    display: flex;
    flex: 1;
    flex-direction: row;
    background-size: cover;
    background-attachment: fixed;
    background-position-x: 50%;
    background-position-y: 50%;

    .particles {
      top: 0;
      right: 0;
      right: 0;
      bottom: 0;
      left: 0;
      z-index: -1;
      display: block;
      position: absolute;
    }
  }

  .version {
    text-align: center;
  }
}
