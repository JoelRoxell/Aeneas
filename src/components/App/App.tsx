import * as React from 'react'
import { Suspense, lazy } from 'react'
import { Switch, Redirect, Route } from 'react-router-dom'
import posed, { PoseGroup } from 'react-pose'

import ProtectedRoute from 'components/common/ProtectedRoute'
import Loader from 'components/common/Loader'

import UserProvider from 'contexts/user-context'
import config from 'project-config'

import * as style from './App.scss'

const Register = lazy(() =>
  import(
    /* webpackPrefetch: true, webpackChunkName: "register" */
    './subs/Register'
  )
)
const Account = lazy(() =>
  import(
    /* webpackPrefetch: true, webpackChunkName: "account" */
    './subs/Account'
  )
)
const Login = lazy(() =>
  import(
    /* webpackPrefetch: true, webpackChunkName: "login" */
    'components/App/subs/Login'
  )
)
const About = lazy(() =>
  import(
    /* webpackPrefetch: true, webpackChunkName: "login" */
    'components/App/subs/About'
  )
)
const SignOut = lazy(() =>
  import(
    /* webpackPrefetch: true, webpackChunkName: "sign-out" */
    'components/App/subs/SignOut'
  )
)

const RouteContainer = posed.div({
  enter: { opacity: 1, delay: 100, beforeChildren: true },
  exit: { opacity: 0 }
})

function App() {
  React.useEffect(() => {
    console.info(`project env: ${process.env.PROJECT_ENV}`)
    console.info(`version: ${config.version}`)
  }, [])

  return (
    <UserProvider>
      <Route
        render={({ location }) => (
          <div className={style.app}>
            <div className={style.view}>
              <PoseGroup>
                <RouteContainer key={location.pathname} style={{ width: '100%' }}>
                  <Suspense fallback={<Loader key="loader" />}>
                    <Switch location={location} key="switcher">
                      <ProtectedRoute
                        path="/account"
                        redirect="/login"
                        component={Account}
                        key="login"
                      />

                      <ProtectedRoute
                        path="/login"
                        redirect="/account"
                        component={Login}
                        reversed
                        key="login"
                      />

                      <ProtectedRoute
                        path="/register"
                        redirect="/account"
                        component={Register}
                        reversed
                        key="register"
                      />

                      <ProtectedRoute
                        path="/sign-out"
                        redirect="/login"
                        component={SignOut}
                        key="sign-out"
                      />

                      <Route path="/about" component={About} key="account" />

                      <Route exact path="/" render={() => <Redirect to="/login" key="none" />} />
                    </Switch>
                  </Suspense>
                </RouteContainer>
              </PoseGroup>
            </div>
          </div>
        )}
      />

      <div>{config.version}</div>
    </UserProvider>
  )
}

export default App
