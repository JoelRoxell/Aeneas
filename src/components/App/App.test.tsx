import * as React from 'react'
import { shallow } from 'enzyme'
import App from 'components/App'

describe('App', () => {
  test('render', () => {
    const wrapper = shallow(<App />)

    expect(wrapper).toMatchSnapshot()
  })
})
