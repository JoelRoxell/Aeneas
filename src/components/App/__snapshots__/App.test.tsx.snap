// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render 1`] = `
<UserProvider>
  <BrowserRouter>
    <div
      className="app"
    >
      <div
        className="view"
      >
        <Suspense
          fallback={<Loader />}
        >
          <Switch>
            <ProtectedRoute
              component={
                Object {
                  "$$typeof": Symbol(react.lazy),
                  "_ctor": [Function],
                  "_result": null,
                  "_status": -1,
                }
              }
              path="/login"
              redirect="/account"
              reversed={true}
            />
            <ProtectedRoute
              component={
                Object {
                  "$$typeof": Symbol(react.lazy),
                  "_ctor": [Function],
                  "_result": null,
                  "_status": -1,
                }
              }
              path="/register"
              redirect="/account"
              reversed={true}
            />
            <ProtectedRoute
              component={
                Object {
                  "$$typeof": Symbol(react.lazy),
                  "_ctor": [Function],
                  "_result": null,
                  "_status": -1,
                }
              }
              path="/sign-out"
              redirect="/login"
            />
            <ProtectedRoute
              component={
                Object {
                  "$$typeof": Symbol(react.lazy),
                  "_ctor": [Function],
                  "_result": null,
                  "_status": -1,
                }
              }
              path="/account"
              redirect="/login"
            />
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
          </Switch>
        </Suspense>
      </div>
    </div>
    <div>
      test version
    </div>
  </BrowserRouter>
</UserProvider>
`;
